src/ComputeSharp.D2D1/Shaders/Interop/Helpers/Rectangle64.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Drawing;
+using System.Runtime.CompilerServices;
+
+namespace ComputeSharp.D2D1.Shaders.Interop.Helpers;
+
+/// <summary>
+/// A 64-bit version of the <see cref="Rectangle"/> type.
+/// </summary>
+internal struct Rectangle64
+{
+    /// <summary>
+    /// The left coordinate of the current rectangle.
+    /// </summary>
+    private long x;
+
+    /// <summary>
+    /// The top coordinate of the current rectangle.
+    /// </summary>
+    private long y;
+
+    /// <summary>
+    /// The width of the current rectangle.
+    /// </summary>
+    private long width;
+
+    /// <summary>
+    /// The height of the current rectangle.
+    /// </summary>
+    private long height;
+
+    /// <summary>
+    /// Creates a new <see cref="Rectangle64"/> instance with the specified parameters.
+    /// </summary>
+    /// <param name="x">The left coordinate of the rectangle.</param>
+    /// <param name="y">The top coordinate of the rectangle.</param>
+    /// <param name="width">The width of the rectangle.</param>
+    /// <param name="height">The height of the rectangle.</param>
+    public Rectangle64(long x, long y, long width, long height)
+    {
+        this.x = x;
+        this.y = y;
+        this.width = width;
+        this.height = height;
+    }
+
+    /// <summary>
+    /// Gets the left coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Left
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.x;
+    }
+
+    /// <summary>
+    /// Gets the top coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Top
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.y;
+    }
+
+    /// <summary>
+    /// Gets the right coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Right
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.x + this.width;
+    }
+
+    /// <summary>
+    /// Gets the bottom coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Bottom
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.y + this.height;
+    }
+
+    /// <summary>
+    /// Gets a <see cref="Rectangle64"/> instance representing an empty rectangle.
+    /// </summary>
+    /// <remarks>The returned instance is such that computing the union of it with any othe rectangle yields the other rectangle.</remarks>
```

Comment:
typo ... `any othe rectangle`

Reply:
Also, computing the intersection of `Empty` and anything should result in `Empty`

Reply:
Fixed, thanks! 🙂